qt/scientific_interfaces/Inelastic/BayesFitting/QuasiView.cpp

```diff
@@ -0,0 +1,355 @@
+    return Qt::blue;
+  } else if (colour == "magenta") {
+    return Qt::magenta;
+  } else if (colour == "cyan") {
+    return Qt::cyan;
+  }
+  return QColor();
+}
+
+} // namespace
+
+namespace MantidQt::CustomInterfaces {
+using namespace InterfaceUtils;
+
+QuasiView::QuasiView(QWidget *parent)
+    : QWidget(parent), m_dblManager(new QtDoublePropertyManager()), m_propTree(new QtTreePropertyBrowser()),
+      m_properties(), m_dblEditorFactory(new DoubleEditorFactory()), m_presenter() {
+  m_uiForm.setupUi(parent);
+
+  m_propTree->setFactoryForManager(m_dblManager, m_dblEditorFactory);
+
+  setupFitOptions();
+  setupPropertyBrowser();
+  setupPlotOptions();
+
+  auto eRangeSelector = m_uiForm.ppPlot->addRangeSelector("QuasiERange");
+  connect(eRangeSelector, &MantidWidgets::RangeSelector::minValueChanged, this, &QuasiView::minEValueChanged);
+  connect(eRangeSelector, &MantidWidgets::RangeSelector::maxValueChanged, this, &QuasiView::maxEValueChanged);
+
+  connect(m_uiForm.chkFixWidth, &QCheckBox::toggled, m_uiForm.mwFixWidthDat, &FileFinderWidget::setEnabled);
+  connect(m_uiForm.chkUseResNorm, &QCheckBox::toggled, m_uiForm.dsResNorm, &DataSelector::setEnabled);
+
+  connect(m_uiForm.dsSample, &DataSelector::dataReady, this, &QuasiView::notifySampleInputReady);
+  connect(m_uiForm.dsSample, &DataSelector::filesAutoLoaded, this, &QuasiView::notifyFileAutoLoaded);
+
+  connect(m_uiForm.dsResolution, &DataSelector::dataReady, this, &QuasiView::notifyResolutionInputReady);
+  connect(m_uiForm.dsResolution, &DataSelector::filesAutoLoaded, this, &QuasiView::notifyFileAutoLoaded);
+
+  connect(m_uiForm.cbProgram, static_cast<void (QComboBox::*)(int const)>(&QComboBox::currentIndexChanged), this,
+          &QuasiView::handleProgramChange);
+  connect(m_uiForm.spPreviewSpectrum, static_cast<void (QSpinBox::*)(int const)>(&QSpinBox::valueChanged), this,
+          &QuasiView::notifyPreviewSpectrumChanged);
+
+  connect(m_uiForm.pbPlotPreview, &QPushButton::clicked, this, &QuasiView::notifyPlotCurrentPreview);
+  connect(m_uiForm.pbSave, &QPushButton::clicked, this, &QuasiView::notifySaveClicked);
+  connect(m_uiForm.pbPlot, &QPushButton::clicked, this, &QuasiView::notifyPlotClicked);
+
+  m_uiForm.dsSample->isOptional(true);
+  m_uiForm.dsResolution->isOptional(true);
+  m_uiForm.dsSample->setWorkspaceTypes({"Workspace2D"});
+  m_uiForm.dsResolution->setWorkspaceTypes({"Workspace2D"});
+}
+
+void QuasiView::subscribe(IQuasiPresenter *presenter) { m_presenter = presenter; }
+
+IRunView *QuasiView::getRunView() const { return m_uiForm.runWidget; }
+
+DataSelector *QuasiView::sampleSelector() const { return m_uiForm.dsSample; }
+
+DataSelector *QuasiView::resolutionSelector() const { return m_uiForm.dsResolution; }
+
+DataSelector *QuasiView::resNormSelector() const { return m_uiForm.dsResNorm; }
+
+FileFinderWidget *QuasiView::fixWidthFileFinder() const { return m_uiForm.mwFixWidthDat; }
+
+void QuasiView::setupFitOptions() {
```

Comment:
This method (and a couple of the below ones) seem like logic that could be in the presenter. Is there a reason this was left in the view?